presto-hive/src/main/java/com/facebook/presto/hive/InternalHiveSplit.java

```diff
@@ -228,23 +226,16 @@ public String toString()
         private static final int HOST_ADDRESS_INSTANCE_SIZE = ClassLayout.parseClass(HostAddress.class).instanceSize() +
                 ClassLayout.parseClass(String.class).instanceSize();
 
-        private final long start;
         private final long end;
         private final List<HostAddress> addresses;
 
-        public InternalHiveBlock(long start, long end, List<HostAddress> addresses)
+        public InternalHiveBlock(long end, List<HostAddress> addresses)
         {
-            checkArgument(start <= end, "block end cannot be before block start");
-            this.start = start;
+            checkArgument(end >= 0, "block end must be >= 0");
```

Comment:
I'm hesitant to do this because it's important that when you get the addresses, they are the same addresses as for the block you're looking at.  That means the order of when the start gets incremented compared to when you get the addresses for the block matter, which is error prone. Either that or you create an InternalHiveBlock every time someone wants the current block, which will be multiple copies of the same thing when the block is splittable.

I'm not sure we should do it unless we find the current memory improvements aren't enough.